### PR TITLE
[CPP-80] Make query message and qlhelp more user-friendly.

### PR DESCRIPTION
**Pull request by zlaski**
ql:   "struct aa has 28 fields, which is too many" becomes
       "struct 'aa' has 28 fields; we suggest refactoring to 15 fields or fewer"

qhelp "...is a sign that the class lacks cohesion (i.e., lacks a single purpose)" becomes
          "...is a sign that the class may lack cohesion (i.e., lack a single purpose)"

Both the .ql and .qhelp have been reformatted in Eclipse via  ctrl+shift+F.

NB: I'm getting 'Pending — Contributor License Agreement is not signed yet.' below, even though I did sign it (and am not allowed to sign it again)...

### CODE REVIEW

File: cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.qhelp [hunk 1]
```diff
@@ -4,33 +4,58 @@
...
+            are not shown. Having too many fields in one
+            class is a sign that the
+            class may lack cohesion (i.e. lack a single
+            purpose).
+            These classes can be split into smaller, more cohesive
+            classes.
+            Alternatively, the related fields can be grouped
+            into
+            <code>struct</code>
+            s.
```
Comment (jbj) [inline cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.qhelp]: This XML auto-formatting doesn't make the file more readable in my opinion, and it also inserts whitespace after the `</code>` tags. Please change the formatting back to what it was before.

File: cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.ql [hunk 2]
```diff
@@ -89,26 +103,33 @@ class ExtClass extends Class {
...
+      )
+    ) and
+  n > 15 and
+  not c.isConstructedFrom(_) and
+  c = vdg.getClass() and
+  if c.hasOneVariableGroup() then suffix = "" else suffix = " - see $@"
+select c,
+  kindstr(c) + " '" + c.getName() + "' has " + n +
+    " fields; we recommend refactoring to 15 fields or fewer" + suffix + ".", vdg,
+  vdg.describeGroup()
```
Comment (jbj) [inline cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.ql]: The autoformat changes in this file look good to me, but please do the autoformat in a separate commit from the real changes. Otherwise it becomes hard to follow the history of this file in the future.

**Comment by geoffw0** [comment]
I think the query message and qhelp changes are an improvement. :+1:

File: cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.qhelp (hunk 1 above)
Comment (zlaski) [reply cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.qhelp]: Will do.  Closing this PR.

**Comment by geoffw0** [comment]
> Closing PR as per jbj's comments.

We generally find it acceptable to edit an existing pull request in response to comments, including removing existing commits and adding new ones, as long as it's clear what's changed.  Keeping all the comments visible at the same URL makes it easier to follow the discussion.
